js/pathfinding.js

```diff
@@ -191,14 +270,33 @@ function calculatePath(from, to, cachedNodes, token, previousWaypoints) {
 			const neighbor = {
 				node: neighborNode,
 				cost: currentNode.cost + edge.cost,
-				estimated: currentNode.cost + edge.cost + estimateCost(neighborNode, from),
+				estimated: currentNode.cost + edge.cost + estimateCost(neighborNode, to),
 				previous: currentNode
 			};
 			nextNodes.pushWithPriority(neighbor);
 		}
 	}
 }
 
+/**
+ * Now we've found the path, we know the final node, and each node links to the previous one.
+ * Reverse this list and return the first node in the path, with each node linking to the next
+ */
+function buildPathNodes(lastNode) {
+	let currentNode = lastNode;
+	let previousNode = null;
+	while (currentNode) {
+		const pathNode = {
+			node: currentNode.node,
+			cost: currentNode.cost,
+			next: previousNode
+		}
+		previousNode = pathNode;
+		currentNode = currentNode.previous;
+	}
+	return previousNode;
+}
```

Comment:
Really simple way of reversing the path, so we can return the first node on the path. This lets us change `previous` to `next` to make it easier to understand what's happening in proceeding code. We also only need to return the fields that we'll use.